mutiny-core/src/nostr/nwc.rs

```diff
@@ -166,99 +185,123 @@ impl NostrWalletConnect {
             let node = node_manager.get_node(from_node).await?;
             if node.get_invoice(&invoice).is_ok_and(|i| i.paid) {
-                return Ok(None);
+                return Ok((None, needs_save));
             }
             drop(node);
 
+            if invoice.amount_milli_satoshis().is_none() {
+                log_error!(
+                    node_manager.logger,
+                    "NWC Invoice amount not set, cannot pay: {invoice}"
+                );
+                return Ok((None, needs_save));
+            }
+
             // if we need approval, just save in the db for later
-            if self.profile.require_approval {
-                let pending = PendingNwcInvoice {
-                    index: self.profile.index,
-                    invoice,
-                    event_id: event.id,
-                    pubkey: event.pubkey,
-                };
-                pending_nwc_lock.lock().await;
-
-                let mut current: Vec<PendingNwcInvoice> = node_manager
-                    .storage
-                    .get_data(PENDING_NWC_EVENTS_KEY)?
-                    .unwrap_or_default();
-
-                current.push(pending);
-                current.sort();
-                current.dedup();
-
-                node_manager
-                    .storage
-                    .set_data(PENDING_NWC_EVENTS_KEY, current, None)?;
-
-                return Ok(None);
-            } else {
-                let msats = invoice.amount_milli_satoshis().unwrap();
-
-                // verify amount is under our limit
-                let content = if msats <= self.profile.max_single_amt_sats * 1_000 {
-                    match self
-                        .pay_nwc_invoice(node_manager, from_node, &invoice)
-                        .await
-                    {
-                        Ok(resp) => resp,
-                        Err(e) => Response {
+            match self.profile.spending_conditions.clone() {
+                SpendingConditions::SingleUse(mut single_use) => {
+                    // check if we have already spent
+                    if single_use.spent {
+                        return Ok((None, needs_save));
+                    }
+
+                    let msats = invoice.amount_milli_satoshis().unwrap();
+
+                    // verify amount is under our limit
+                    let content = if msats <= single_use.amount_sats * 1_000 {
+                        match self
+                            .pay_nwc_invoice(node_manager, from_node, &invoice)
+                            .await
+                        {
+                            Ok(resp) => {
+                                // mark as spent and disable profile
+                                single_use.spent = true;
+                                self.profile.spending_conditions =
+                                    SpendingConditions::SingleUse(single_use);
+                                self.profile.enabled = false;
```

Comment:
thoughts on deleting the profile instead? Don't need many profiles sitting around if this is a popular thing.

Reply:
If we delete then keys will be reused, maybe we should add another flag like `archived` so it doesn't show up in the list

Reply:
Damn, well maybe we need to solve the key reuse thing as part of this too. Maybe we blow away or rewrite all profiles and communicate to user's that they need to relink them?

Reply:
If we want to change how we derive keys for nwc, probably suitable for a follow up (would be a pretty large change)

Reply:
Good point. Then set an archive flag with a TODO comment? Not sure, users have asked to remove some NWCs anyways not just gifting ones. Up to you.

Reply:
Okay added the archiving